Add ability to submit topology to the specific YARN queue on YARN mode #1427

mycFelix opened the pull request:
Heron supports submitting topology to the YARN `default` queue now. I was wondering if Heron should support submitting topology to the specific YARN queue? I try to use the `--config-property` arg to implements this function.

if user wants to submit topology to YARN with the `test` queue. The submitting command could be
`heron submit yarn ~/.heron/examples/heron-examples.jar com.twitter.heron.examples.ExclamationTopology ExclamationTopology --deploy-deactivated --config-property heron.scheduler.yarn.queue=test`

And the results would be
![image](https://cloud.githubusercontent.com/assets/4719414/18825567/b19aebb0-83fb-11e6-958b-87ee566eb4a7.png)

This PR fixes #1427 and the changes are following:
* added `heron.scheduler.yarn.queue` args in the `com.twitter.heron.spi.common.keys.yaml` file
* supports submitting topology to the specific yarn queue by parsing configs in the `com.twitter.heron.scheduler.yarn.YarnLauncher` class.

ashvina commented on heron/spi/src/java/com/twitter/heron/spi/common/keys.yaml:
```diff
@@ -55,6 +55,7 @@ STATE_MANAGER_CLASS:                 heron.class.state.manager
 # keys for scheduler config
 ################################################################################
 SCHEDULER_IS_SERVICE:                heron.scheduler.is.service
+HERON_SCHEDULER_YARN_QUEUE:          heron.scheduler.yarn.queue
```
Would it be useful to update the default value in the `yarn/scheduler.yaml` too. When needed users could find it in the config file and avoid searching for it. Also could you please add this config in the document.

ashvina commented:
Two small suggestions. Let me know what you think.

1. You could link this PR with the issue by including `Fixes ISSUE_NUMBER` in the description of this PR.
1. It would be better to `git squash` the commits to avoid confusion and keep commit history readable.

mycFelix commented:
@ashvina - Thank you so much for your suggestions. I linked this PR with issue number and tried to reform the commits. I'll send another PR for updating `yarn/scheduler.yaml` and the doc after this PR is merged. Please let me know if there's anything else needed.

mycFelix commented on heron/spi/src/java/com/twitter/heron/spi/common/Context.java:
```diff
@@ -346,4 +346,8 @@ public static final String stateManagerSandboxClassPath(Config cfg) {
     return cfg.getStringValue(ConfigKeys.get("SANDBOX_UPLOADER_CLASSPATH"));
   }
+
+  public static final String heronYarnQueue(Config cfg) {
```
Planning to move this method to the YarnContext. I'll change it~

mycFelix commented:
@ashvina - I added `YarnContext` class, updated the  `conf/yarn/scheduler` file and the doc. Could you please look at this change again? Thank you very much for taking your time.

kramasamy commented:
@ashvina - is this good to go?

ashvina commented:
LGTM 👍
